cmd/kube-apiserver/app/server.go

```diff
@@ -379,6 +380,15 @@ func CreateKubeAPIServerConfig(
 	if config.GenericConfig.EgressSelector != nil {
 		// Use the config.GenericConfig.EgressSelector lookup to find the dialer to connect to the kubelet
 		config.ExtraConfig.KubeletClientConfig.Lookup = config.GenericConfig.EgressSelector.Lookup
+
+		// Use the config.GenericConfig.EgressSelector lookup as the proxyTransport dialer
+		networkContext := egressselector.Cluster.AsNetworkContext()
+		dialer, err := config.GenericConfig.EgressSelector.Lookup(networkContext)
+		if err != nil {
+			return nil, nil, nil, nil, err
+		}
+		// override the contextual dialer to use konnectivity server which is used over UDS
+		proxyTransport.DialContext = dialer
```

Comment:
proxyTransport is of type *http.Transport, so changing it this way may have side effects.

How about this:
```golang
		networkContext := egressselector.Cluster.AsNetworkContext()
		dialer, err := config.GenericConfig.EgressSelector.Lookup(networkContext)
		if err != nil {
			return nil, nil, nil, nil, err
		}
                c := proxyTransport.Clone()
                c.DialContext = dialer
                // this only affects the transport used by the "proxy" subresources.
                config.ExtraConfig.ProxyTransport = c
```

Reply:
@caesarxuchao looks good, updated :) I'll give it a test today or tomorrow

Reply:
And it works as well :D